pilot/pkg/bootstrap/config_compare.go

```diff
@@ -46,6 +46,16 @@ func needsPush(prev config.Config, curr config.Config) bool {
 			return true
 		}
 	}
+	for label := range prev.Meta.Labels {
+		if strings.Contains(label, "istio.io") {
+			return true
+		}
+	}
+	for annotation := range prev.Meta.Annotations {
+		if strings.Contains(annotation, "istio.io") {
```

Comment:
I donot understand why we push when labels and annotations are equal even both with `istio.io`

Reply:
There was long discussion on this https://github.com/istio/istio/pull/34312 and we settled on this logic. So not sure if it is safe to change that now - But this PR is a definite bug fix.

Reply:
One of the main reasons why we do not compare labels is if unrelated labels change we do not want to trigger push for configs like DR - as mentioned in https://github.com/istio/istio/issues/33600